### PR TITLE
add some rust test case 

### PR DESCRIPTION
**Pull request by carltraveler**

### CODE REVIEW

File: wasmtest/contracts-rust/contractop/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
+#![cfg_attr(not(feature = "mock"), no_std)]
+#![feature(proc_macro_hygiene)]
```
Comment (laizy) [inline wasmtest/contracts-rust/contractop/src/lib.rs]: this can be removed.

File: wasmtest/contracts-rust/contractop/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+#[no_mangle]
+pub fn invoke() {
+    let input = runtime::input();
+    let mut source = ZeroCopySource::new(&input);
+    let action: &[u8] = source.read().unwrap();
+    let mut sink = Sink::new(12);
+    match action {
+        b"add" => {
+            let (a, b) = source.read().unwrap();
```
Comment (laizy) [inline wasmtest/contracts-rust/contractop/src/lib.rs]: remove this function? not used in the following test case.

File: wasmtest/contracts-rust/contractop/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+            if let Some(address) =
+                runtime::contract_migrate(code, 3, "name", "version", "author", "email", "desc")
+            {
+                if let Some(resv) = runtime::call_contract(&address, &[]) {
+                    let mut source = ZeroCopySource::new(&resv);
+                    let val: u64 = source.read().unwrap();
+                    assert!(val == 0x32733);
+                } else {
+                    panic!("call migrate failed");
```
Comment (laizy) [inline wasmtest/contracts-rust/contractop/src/lib.rs]: you can use `runtime::call_contract(&address, &[]).expect("call migrate failed")`  to simplify this.

File: wasmtest/contracts-rust/contractop/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            let mut sink = Sink::new(8);
+            let key: [u8; 1] = [0x25];
+            let val: u64 = 0x138297;
+            sink.write(val);
+            runtime::storage_write(&key, sink.bytes());
+            if let Some(val) = runtime::storage_read(&key) {
+                let mut ressource = ZeroCopySource::new(&val);
+                let res: u64 = ressource.read().unwrap();
+                assert!(res == 0x138297);
+            } else {
```
Comment (laizy) [inline wasmtest/contracts-rust/contractop/src/lib.rs]: ditto.

File: wasmtest/contracts-rust/contractop/src/lib.rs (hunk 3 above)
Comment (carltraveler) [reply wasmtest/contracts-rust/contractop/src/lib.rs]: then how to handle the val assert.

**Review by laizy — APPROVED** [review]

**Pull request merged**